server/boards/routes.ts

```diff
@@ -686,4 +688,87 @@ router.delete(
   }
 );
 
+
+/**
+ * @openapi
+ * /boards/{board_id}/roles:
+ *   get:
+ *     summary: Fetches latest roles summary for the board.
+ *     operationId: getBoardRolesByExternalId
+ *     tags:
+ *       - /boards/
+ *     security:
+ *       - {}
+ *     parameters:
+ *       - name: board_id
+ *         in: path
+ *         description: The id of the board.
+ *         required: true
+ *         schema:
+ *           type: string
+ *           format: uuid
+ *     responses:
+ *       200:
+ *         description: The board roles summary.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: "#/components/schemas/RealmRoles"
+ *             examples:
+ *               twisted_minds:
+ *                 value:
+ *                   roles:
+ *                     - user_id: "1"
+ *                       username: "bobatan"
+ *                       role_id: "3"
+ *                       role_name: "The Owner"
+ *       401:
+ *         $ref: "#/components/responses/ensureLoggedIn401"
+ *       403:
+ *         $ref: "#/components/responses/ensurePermission403"
+ *       404:
+ *         description: The board was not found.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: "#/components/schemas/genericResponse"
+ *       500:
+ *         description: There was an error fetching board roles.
+ */
+
+router.get(
+  "/:board_id/roles",
+  ensureBoardAccess, ensureLoggedIn,
+  ensureBoardPermission(BoardPermissions.viewRolesOnBoard),
+  async (req, res) => {
+	  try {
+	    const { board_id } = req.params;
+	    const boardRoles = await getBoardRoles({
+	      boardExternalId: board_id,
+	    });
+			if (!boardRoles?.length){
+				res.status(200).json({roles:[]});
+				return;
+			}
+			const formattedBoardRoles = boardRoles.map((roleEntry) => {
```

Comment:
You can probably get this formatting directly from the SQL query. Let me know if you need help figuring it out!